### PR TITLE
Ukrainian and Russian localization of the OCS

### PR DESCRIPTION
**Pull request by semteacher**
I am forked of the original Open Conference System repository and extend it with Ukrainian locale support (and Russian too)

Key changes
Added 
Ukrainian locale from https://github.com/semteacher/ocs_in_Ukrainian
Russian locale from https://github.com/pkp/ocs/pull/8
Ukrainian / Russian blocks and plug-in locales from OJS repositoriy https://github.com/pkp/ojs

Updated
Ukrainian locale in master branch up to 2015 may
Ukrainian / Russian blocks and plug-in locales

### CODE REVIEW

File: plugins/importexport/native/locale/ru_RU/locale.xml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	<message key="plugins.importexport.native.import.error.paperAbstractLocaleUnsupported">A paper abstract was provided for the paper "{$paperTitle}" in a locale ("{$locale}") that this conference does not support.</message>
+	<message key="plugins.importexport.native.import.error.galleyLabelMissing">Статье "{$articleTitle}" в секции "{$sectionTitle}" выпуска "{$issueTitle}" не хватает метки гранок.</message>
+	<message key="plugins.importexport.native.import.error.suppFileMissing">Статье "{$articleTitle}" в секции "{$sectionTitle}" выпуска "{$issueTitle}" не хватает дополнительного файла.</message>
+	<message key="plugins.importexport.native.import.error.galleyFileMissing">Статье "{$articleTitle}" в секции "{$sectionTitle}" выпуска "{$issueTitle}" не хватает файла гранок.</message>
+
+	<message key="plugins.importexport.native.import.error.trackTitleLocaleUnsupported">A track title ("{$trackTitle}") was provided in a locale ("{$locale}") that this conference does not support.</message>
```
Comment (mtub) [inline plugins/importexport/native/locale/ru_RU/locale.xml]: Please remove the lines you haven't translated (81-96)

File: plugins/importexport/native/locale/uk_UA/locale.xml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+{$scriptName} {$pluginName} import [xmlFileName] [journal_path] [user_name]
+       issue_id [issueId] section_id [sectionId]
+
+{$scriptName} {$pluginName} import [xmlFileName] [journal_path] [user_name]
+       issue_id [issueId] section_name [name]
+
+{$scriptName} {$pluginName} import [xmlFileName] [journal_path]
+       issue_id [issueId] section_abbrev [abbrev]]]></message>
+	<message key="plugins.importexport.native.export">Експорт даних</message>
+	<message key="plugins.importexport.native.export.tracks">Export Tracks</message>
```
Comment (mtub) [inline plugins/importexport/native/locale/uk_UA/locale.xml]: not translated…

File: plugins/importexport/native/locale/uk_UA/locale.xml [hunk 3]
```diff
@@ -0,0 +1,113 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE locale SYSTEM "../../../../../lib/pkp/dtd/locale.dtd">
+<!--
+  * locale.xml
+  *
+  * Copyright (c) 2000-2012 John Willinsky
+  * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+  *
+  * Localization strings for the en_US (U.S. English) locale.
```
Comment (mtub) [inline plugins/importexport/native/locale/uk_UA/locale.xml]: wrong locale name

File: plugins/importexport/native/locale/ru_RU/locale.xml [hunk 4]
```diff
@@ -0,0 +1,112 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE locale SYSTEM "../../../../../lib/pkp/dtd/locale.dtd">
+<!--
+  * locale.xml
+  *
+  * Copyright (c) 2000-2012 John Willinsky
+  * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+  *
+  * Localization strings for the en_US (U.S. English) locale.
```
Comment (mtub) [inline plugins/importexport/native/locale/ru_RU/locale.xml]: wrong locale name

File: plugins/importexport/mets/locale/ru_RU/locale.xml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  * plugins/importexport/mets/locale/ru_RU/locale.xml
+  *
+  * Copyright (c) 2003-2013 John Willinsky
+  * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+  *
+  * Localization strings for the en_US (U.S. English) locale.
+  *
+  -->
+ 
+<locale name="en_US" full_name="U.S. English">
```
Comment (mtub) [inline plugins/importexport/mets/locale/ru_RU/locale.xml]: wrong locale name

File: plugins/importexport/mets/locale/ru_RU/locale.xml [hunk 6]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE locale SYSTEM "../../../../../lib/pkp/dtd/locale.dtd">
+
+<!--
+  * plugins/importexport/mets/locale/ru_RU/locale.xml
+  *
+  * Copyright (c) 2003-2013 John Willinsky
+  * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+  *
+  * Localization strings for the en_US (U.S. English) locale.
```
Comment (mtub) [inline plugins/importexport/mets/locale/ru_RU/locale.xml]: wrong locale name

File: plugins/citationFormats/turabian/locale/ru_RU/locale.xml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  * plugins/citationFormats/turabian/locale/ru_RU/locale.xml
+  *
+  * Copyright (c) 2003-2013 John Willinsky
+  * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+  *
+  * Localization strings for this citation plugin in en_US.
+  *
+  -->
+ 
+<locale name="en_US" full_name="U.S. English">
```
Comment (mtub) [inline plugins/citationFormats/turabian/locale/ru_RU/locale.xml]: wrong locale name

File: locale/uk_UA/admin.xml [hunk 8]
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE locale SYSTEM "../../lib/pkp/dtd/locale.dtd">
+<!--
+  * ocs2-devel/locale/en_US/admin.xml
+  *
+  * Copyright (c) 2000-2012 John Willinsky
+  * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+  *
+  * Localization strings for the en_US (English) locale.
```
Comment (mtub) [inline locale/uk_UA/admin.xml]: wrong locale name

File: locale/uk_UA/admin.xml [hunk 9]
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE locale SYSTEM "../../lib/pkp/dtd/locale.dtd">
+<!--
+  * ocs2-devel/locale/en_US/admin.xml
```
Comment (mtub) [inline locale/uk_UA/admin.xml]: wrong locale name

File: locale/uk_UA/admin.xml [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+  -->
+<locale name="uk_UA" full_name="Ukraine">
+	<message key="admin.languages.downloadUnavailable"><![CDATA[<p>Downloading language packs from the Public Knowledge Project web server is not currently available because:</p>
+	<ul>
+		<li>Your server does not have or allow execution of the GNU "tar" utility</li>
+		<li>OCS is unable to modify the locale registry file, typically "registry/locales.xml".</li>
+	</ul>
+<p>Language packs can be manually downloaded from the <a href="http://pkp.sfu.ca">PKP web site</a>.</p>]]></message>
+
+	<message key="admin.languages.confirmDisable">Are you sure you want to disable this locale? This may affect any hosted conferences currently using the locale.</message>
```
Comment (mtub) [inline locale/uk_UA/admin.xml]: not translated

File: locale/uk_UA/author.xml [hunk 11]
```diff
@@ -0,0 +1,129 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE locale SYSTEM "../../lib/pkp/dtd/locale.dtd">
+<!--
+  * ocs2-devel/locale/en_US/author.xml
```
Comment (mtub) [inline locale/uk_UA/author.xml]: wrong locale name

File: locale/uk_UA/author.xml [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+  *
+  * Copyright (c) 2000-2012 John Willinsky
+  * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+  *
+  * Localization strings for the en_US (English) locale.
```
Comment (mtub) [inline locale/uk_UA/author.xml]: wrong locale name

File: registry/locales.xml [hunk 13]
```diff
@@ -27,4 +27,6 @@
 	<locale key="pt_BR" name="Português (Brasil)" iso639-2b="por" iso639-3="por" />
 	<locale key="pt_PT" complete="false" name="Português (Portugal)" iso639-2b="por" iso639-3="por" />
 	<locale key="zh_TW" complete="false" name="正體中文" iso639-2b="chi" iso639-3="zho" />
+    <locale key="uk_UA" complete="false" name="Українська" iso639-2b="ukr" iso639-3="ukr"/>
```
Comment (mtub) [inline registry/locales.xml]: indentation... (30/31)

**Comment by mtub** [comment]
Thanks, @semteacher . 

I'm not sure I understand how you've produced the files but most of it looks good.

I've commented on a few issues, mostly lines that haven't been translated. I recommend that you either complete the installation or remove those lines. Also, please check the name of the locale in the files header and beginning; a few times you have included "en_US" where it should be uk_UK or ru_RU, etc. 

Do you mind removing three files from the pull request? 

These two may be useful in your fork but don't belong into the PKP repository:
* README.md
* localesync.php

Furthermore, we shouldn't treat changes of specifications together with locale changes. If you want to request that the conference title field should accept longer inputs then please file another pull request or bug report and restore 
* templates/manager/setup/step1.tpl.

Thanks for your contribution!

**Comment by semteacher** [comment]
Dear @mtub 
I try to fix all issues that you was wrote about it. Hope now this localization will have chance to be included into the OCS codebase...

**Comment by mtub** [comment]
@semteacher, are you actually running code from the OCS master branch? 

I am asking because the ocs-dev-2_3 branch (https://github.com/pkp/ocs/tree/ocs-dev-2_3) already has a Ukrainian locale. If you didn't see the existing Ukrainian locale (it was only recently added to the ocs-dev-2_3 branch, see #13) I am very sorry for your extra work. I apologize that I didn't realize this before asking you to fix things in this pull request, as we should have talked about what to do with the additional translation before going into that kind of detail.

The OCS code is organized in branches, and the next OCS release probably will be based on the ocs-dev-2_3 branch. But since you translated the (older) master branch I was wondering if you are running that code somewhere.

Having the master branch translated isn't a problem in itself (there remain a few minor issues with your latest commits that can be easily fixed), but the code isn't used at the moment. Furthermore, I am not so comfortable with having different translations of the same English text in the two OCS branches. Did you look at the Ukrainian locale in the ocs-dev-2_3 branch? I've just run a quick comparison, and it didn't look as if you've used that code.

**Comment by semteacher** [comment]
Hello!
Hm.... I am very bad surprised one side. But I am happy that I am not alone
in Ukraine who disturb on localization supporting, maintaining and (most
important) sharing.

It is very bad for my that I was not accurately consider ocs-dev-2_3 branch
and miss #13 <https://github.com/pkp/ocs/pull/13>. I pay for this a lot of
work.... It will be experience for me...

I translate "master" branch based on my previous localization experience
with LMS Moodle. There each new release every time based on the "master"
branch...
Practically I run OCS 2.3.6  in our university to support conferences...
And I have translated it too
https://github.com/semteacher/ocs/tree/ocs-2_3_6-ua_locale_new

Currently I did not look at the Ukrainian locale in the ocs-dev-2_3 branch.
I will review it and try to compare with my version...


On Fri, Jun 12, 2015 at 9:27 AM, Marco Tullney <notifications@github.com>
wrote:

> @semteacher <https://github.com/semteacher>, are you actually running
> code from the OCS master branch?
>
> I am asking because the ocs-dev-2_3 branch (
> https://github.com/pkp/ocs/tree/ocs-dev-2_3) already has a Ukrainian
> locale. If you didn't see the existing Ukrainian locale (it was only
> recently added to the ocs-dev-2_3 branch, see #13
> <https://github.com/pkp/ocs/pull/13>) I am very sorry for your extra
> work. I apologize that I didn't realize this before asking you to fix
> things in this pull request, as we should have talked about what to do with
> the additional translation before going into that kind of detail.
>
> The OCS code is organized in branches, and the next OCS release probably
> will be based on the ocs-dev-2_3 branch. But since you translated the
> (older) master branch I was wondering if you are running that code
> somewhere.
>
> Having the master branch translated isn't a problem in itself (there
> remain a few minor issues with your latest commits that can be easily
> fixed), but the code isn't used at the moment. Furthermore, I am not so
> comfortable with having different translations of the same English text in
> the two OCS branches. Did you look at the Ukrainian locale in the
> ocs-dev-2_3 branch? I've just run a quick comparison, and it didn't look as
> if you've used that code.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pkp/ocs/pull/15#issuecomment-111377103>.
>



-- 
З повагою, Семенець А.В.

**Comment by mtub** [comment]
@semteacher, I am very sorry! Let me just add that your help is really appreciated and that this will lead to us reviewing the current documentation and processes to minimize the risk of volunteers to run into the same situation.

**Pull request closed without merging**